tools/tinytest-codegen.py

```diff
@@ -112,11 +130,20 @@ def script_to_map(test_file):
 )
 argparser.add_argument("--stdin", action="store_true", help="read list of tests from stdin")
 argparser.add_argument("--exclude", action="append", help="exclude test by name")
+argparser.add_argument(
```

Comment:
We should use `argparse.FileType` in other scripts! I didn't realise that it was supported in Python 2.x but it is. (No action required)

Reply:
I missed the part where Python 2.x support was a requirement.  I picked the minimum syntax version from what is ran in the CI (3.8).  In this case the script as it is will fail under 2.x as I used the walrus operator in the profile parser...